lib/modules/dosomething/dosomething_api/resources/signup_resource.inc

```diff
@@ -189,6 +189,25 @@ function _signup_resource_index($user, $users, $campaigns, $count, $page, $compe
     'runs' => $runs,
   ];
 
+  if (variable_get('rogue_collection', FALSE)) {
+    if ($user) {
+      $northstar_user_id = dosomething_user_get_northstar_id($user->uid);
+    }
+
+     $rogue = dosomething_rogue_client();
+
+     $response = $rogue->getActivity([
+       'filter' => [
+         'northstar_id' => $northstar_user_id,
+         'campaign_id' => $campaigns[0],
+       ],
+     ]);
+
+     if ($response) {
+      return $response;
```

Comment:
I think the approach I'd recommend is transforming the Rogue request inline right here, so like:

```php

$transformed_response = collect($response['data'])->map(function($signup) {
   return [
    'id' => $signup->id,
    'created_at' => new Carbon($signup->created_at)->timestamp,
    'campaign_run' => [
      'id' => $signup->campaign_run_id,
      // ... and so on
});

```